### PR TITLE
Release 24.1.2 to noble

### PR DESCRIPTION
**Pull request by blackboxsw**
We are specifically avoiding merging this to ubuntu/devel as Ubuntu is in feature freeze at the moment and we only want to upload bugfix releases to ubuntu/noble-24.1.x.

## Branch creation process 
- Updated hotfix process card https://github.com/orgs/canonical/projects/29/views/1?pane=issue&itemId=23587201

```
git fetch upstream --tags
git checkout upstream/ubuntu/noble-24.1.x -B ubuntu/noble-24.1.x
new_upstream_snapshot -c 24.1.2  # expect merge failures on Changelog && cloudinit/version.py
vi Changelog cloudinit/version.py # Accept new 24.1.2 content
git add ChangeLog cloudinit/version.py
git merge --continue
# continue with new_upstream_snapshot post-merge
new_upstream_release.py -c 24.1.2 -p merge # proceed with refreshing debian/changelog entry creation post merge

# manually correct debian/changelog
# 1. package version 24.1.2-0ubuntu1 instead of 0ubuntu0
# 2. change "Upstream snapshot based on" -> "Upstream bugfix release based on" for clarity during feature freeze
# 3. drop duplicated bugs 2056439, 2056460, 2055077 2056194 referenced in latest change 24.1.2 change log entry
vi debian/changelog
git commit -am 'update changelog'
# squash commits to changelog 'fixup' the latest into the previous commit
git rebase -u HEAD~2
dch -r -D noble
git commit -m 'releasing cloudinit version 24.1.2-0ubuntu1' debian/changelog
```
## Merge type

- [ ] Squash merge using "Proposed Commit Message"
- [x] Rebase and merge unique commits. Requires commit messages per-commit each referencing the pull request number (#<PR_NUM>)

### CODE REVIEW

**Review by aciba90 — APPROVED** [review]
LGTM!

File: debian/changelog [hunk 1]
```diff
@@ -1,3 +1,12 @@
+cloud-init (24.1.2-0ubuntu0) UNRELEASED; urgency=medium
```
Comment (TheRealFalcon) [inline debian/changelog]: > 1. package version 24.1.2-0ubuntu1 instead of 0ubuntu0

Whoops, look like this didn't happen.

**Review by TheRealFalcon — CHANGES_REQUESTED** [review]
Version number needs changing, but otherwise LGTM.

It is odd that git had replay all of 24.1.1 here instead of only merging in 24.1.2. I'm not really sure why, but when doing a `git diff --name-only 24.1.2`, I only see changes in the debian directory, so it should fine.

**Review by TheRealFalcon — APPROVED** [review]
LGTM!

**Pull request merged**